script.js

```diff
@@ -1,4 +1,233 @@
+// the user inputs one of "scissors", "paper", or "stone",
+// the program internally randomly chooses scissors, paper, or stone,
+// and the program outputs whether the user won, the program won, or it's a draw.
+
+// global variables
+var gameMode = "waiting for username";
+var username = "";
+var playCount = 0;
+var winCount = 0;
+var mjpWin = "no one"; // korean SPS winner
+
+// SPS function randomises the computer choice
+function SPS() {
+  var randomiser = Math.floor(Math.random() * 3); // using randomiser to pick a number
+  var computer = ["scissors", "paper", "stone"];
+  return computer[randomiser]; // returning the results of the randomised choice
+}
+
+// ==================== GAME MODE ======================
+function gamePlay(gameMode, comPlay, input) {
+  if (gameMode == "playing SPS") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = playSPS(comPlay, input);
+  }
+  if (gameMode == "Reversed SPS") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = reversePlay(comPlay, input);
+  }
+  if (gameMode == "Muk-Jji-Pa!") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = koreanSPS(comPlay, input);
+  }
+  if (gameMode == "Computer vs Computer") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = comVcom(comPlay, input);
+  }
+  return myOutputValue;
+}
+
+// ==================== NORMAL SPS ======================
+function playSPS(comPlay, input) {
+  myOutputValue =
+    "Hello " +
+    username +
+    ". <br>Playing Scissors-Paper-Stone. <br>Enter scissors, paper or stone to play.";
+  console.log("game: " + gameMode);
+
+  // if player chose the same as com
+  if (input == comPlay) {
+    // +1 play count each time the game runs
+    playCount += 1;
+    myOutputValue = `You played ${input}. <br> Computer played ${comPlay}. <br><br>It's a draw! Play again?<br><br>${username} has won ${winCount}/${playCount} times against the computer.`;
+  } else if (
+    // else if player wins computer
+    (input == "scissors" && comPlay == "paper") ||
+    (input == "paper" && comPlay == "stone") ||
+    (input == "stone" && comPlay == "scissors")
+  ) {
+    // +1 play count each time the game runs
+    playCount += 1;
+    // +1 win when user wins
+    winCount += 1;
+    myOutputValue = `You played ${input}. <br> Computer played ${comPlay}. <br><br>You win! Play again?<br><br>${username} has won ${winCount}/${playCount} times against the computer.`;
+  } else if (
+    // else if player loses
+    (input == "scissors" && comPlay == "stone") ||
+    (input == "paper" && comPlay == "scissors") ||
+    (input == "stone" && comPlay == "paper")
+  ) {
+    // +1 play count each time the game runs
+    playCount += 1;
+    myOutputValue = `You played ${input}. <br> Computer played ${comPlay}. <br><br>You lose! Try again?<br><br>${username} has won ${winCount}/${playCount} times against the computer.`;
+  }
+  return myOutputValue;
+}
```

Comment:
Since playCount increases by 1 in every if statement, consider putting it before the if statements to reduce redundancy.